vars/skipStage.groovy

```diff
@@ -10,6 +13,74 @@
  * Method to return true or false if a commit pragma says to skip a stage
  */
 
+// Determine if this is a restarted job and if it succeeded on the
+// previous run.
+boolean already_passed() {
+    // Lookup to see if restarted
+    String restart_cause = 'org.jenkinsci.plugins.pipeline.modeldefinition.' +
+                   'causes.RestartDeclarativePipelineCause'
+
+    def restarted = currentBuild.getBuildCauses().any { cause ->
+        cause._class == restart_cause }
+    if (!restarted) {
+        return false
+    }
+
+    // Make sure a previous check has been removed.
+    String status_file = stageStatusFilename()
+    if (fileExists(status_file)) {
+        fileOperations([fileDeleteOperation(includes: status_file)])
+    }
+
+    String old_build = "${env.BUILD_NUMBER.toInteger() - 1}"
+
+    // First try looking up using copyArtifacts, which requires
+    // that the Jenkinsfile give permission for the copy.
+    try {
+        copyArtifacts projectName: env.JOB_NAME,
+                      optional: false,
+                      filter: status_file,
+                      selector: specific(old_build)
+        try {
+            String stage_status = readFile(file: status_file)
+            println("stage ${env.STAGE_NAME} status '${stage_status}'")
+            if (stage_status == 'SUCCESS') {
+                return true
+            }
```

Comment:
Or maybe better yet, move the message here and add to it that the stage is rerunning:

```suggestion
            }
            println("Previous run, this stage ended with status '${stage_status}'", so rerunning)
```

Reply:
Will be in next push